app/battle/battle.py

```diff
@@ -0,0 +1,24 @@
+from typing import Any
+
+
+class Battle:
+    def __init__(self, knight1: Any, knight2: Any) -> None:
+        self.knight1 = knight1
+        self.knight2 = knight2
+
+    def perform_battle(self) -> None:
+        print("Fight!")
+        print(f"{self.knight1.name} against {self.knight2.name}")
+
+        while self.knight1.hp > 0 and self.knight2.hp > 0:

```

Comment:
Arsen means that you don't need this `while` loop, hp is not bound to be less than 0 to end battle